sdks/wasm/framework/src/WebAssembly.Bindings/Runtime.cs

```diff
@@ -391,46 +391,30 @@ static object ObjectToEnum (IntPtr method_handle, int parm, object obj)
 
 		}
 
-
-		static MethodInfo gsjsc;
-		static void GenericSetupJSContinuation<T> (Task<T> task, JSObject cont_obj)
-		{
-			task.GetAwaiter ().OnCompleted ((Action)(() => {
-
-				if (task.Exception != null)
-					cont_obj.Invoke ((string)"reject", task.Exception.ToString ());
-				else {
-					cont_obj.Invoke ((string)"resolve", task.Result);
-				}
-
-				cont_obj.Dispose ();
-				FreeObject (task);
-
-			}));
-		}
-
 		static void SetupJSContinuation (Task task, JSObject cont_obj)
 		{
-			if (task.GetType () == typeof (Task)) {
-				task.GetAwaiter ().OnCompleted ((Action)(() => {
-
-					if (task.Exception != null)
-						cont_obj.Invoke ((string)"reject", task.Exception.ToString ());
-					else
-						cont_obj.Invoke ((string)"resolve", (object [])null);
+			if (task.IsCompleted)
+				Complete ();
+			else
+				task.GetAwaiter ().OnCompleted (Complete);
 
+			void Complete () {
+				try {
+					if (task.Exception == null) {
+						var resultProperty = task.GetType ().GetProperty("Result");
+						cont_obj.Invoke ("resolve", resultProperty?.GetValue(task));
```

Comment:
In the old code we were passing a null object[] in the no-result case, now we'll be allocating a 1-element object[] and putting null in it, I think? Maybe not a big deal but it is a behavior change w/GC implications

Reply:
I think this also means that the JS side will get an arguments[] containing 1 element instead of 0, which is an observable behavior change that could cause problems if the code is paying attention to arg count

Reply:
good catch